### PR TITLE
Role creation

### PR DESCRIPTION
**Pull request by astrocbxy**
first role creation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+Sets up and configures a pretix instance.
+
+
+## Requirements
+
+It needs an apt based system like Debian or Ubuntu. Also the [stuvusIT.nginx](https://github.com/stuvusIT/nginx), [stuvusIT.redis](https://github.com/stuvusIT/redis) and [stuvusIT.postgresql](https://github.com/stuvusIT/postgresql) roles are required.
+
+
+
```
Comment (SlothOfAnarchy) [inline README.md]: Remove unnecessary whitespace (leave two lines)

File: README.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
+# Pretix
+
+Sets up and configures a pretix instance.
+
+
+## Requirements
+
+It needs an apt based system like Debian or Ubuntu. Also the [stuvusIT.nginx](https://github.com/stuvusIT/nginx), [stuvusIT.redis](https://github.com/stuvusIT/redis) and [stuvusIT.postgresql](https://github.com/stuvusIT/postgresql) roles are required.
```
Comment (SlothOfAnarchy) [inline README.md]: Please split each sentence into one line, this makes it easier to see differences between commits.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+
+
+
+
+## Role Variables
+
+| Name                                | Required                 | Default                                                                     | Description                                                                                                                                                                                     |
+|:------------------------------------|:------------------------:|:----------------------------------------------------------------------------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `pretix_instance_name`                 | :heavy_check_mark:       |                                                                             | Name of the Pretix Installation |
```
Comment (SlothOfAnarchy) [inline README.md]:
```suggestion
| `pretix_instance_name`                 | :heavy_check_mark:       |                                                                             | Name of the Pretix installation |
```

File: README.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+| `pretix_instance_url`                  | :heavy_check_mark:       |                                                                             | Full URL of the installation without trailing Slash |
+| `pretix_mail_address`                  | :heavy_check_mark:       |                                                                             | Name of the mail address, pretix will send mails from |
```
Comment (SlothOfAnarchy) [inline README.md]:
```suggestion
| `pretix_mail_address`                  | :heavy_check_mark:       |                                                                             | Name of the mail address Pretix will send mails from |
```

File: README.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+| `pretix_mail_username`                 | :heavy_check_mark:       |                                                                             | Username of mail account to send mails from |
+| `pretix_mail_password`                 | :heavy_check_mark:       |                                                                             | Password of mail account to send mails from |
+| `pretix_mail_host`                     | :heavy_multiplication_x:       |  `127.0.0.1`                                                          | Full URL of the installation without trailing Slash |
```
Comment (SlothOfAnarchy) [inline README.md]: Wrong description, same for `pretix_mail_port`.

File: README.md [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+| `pretix_mail_port`                     | :heavy_multiplication_x:       |  `465`                                                                | Name of the mail address, pretix will send mails from |
+| `pretix_mail_tls`                      | :heavy_multiplication_x:       |  `on`                                                                 | "on" if you want to authenticata smtp with TLS |
```
Comment (SlothOfAnarchy) [inline README.md]: Please use a boolean value for this and convert it to the `on`/`off` string in the config if needed.

File: README.md [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+| `pretix_user`                      | :heavy_multiplication_x:       |  `pretix`                                                                 | Name of the User pretix will be installed under |
+| `pretix_group`                     | :heavy_multiplication_x:       |  `pretix`                                                                 | Name of the Usergroup pretix will be installed under |
+| `pretix_config_dor`              | :heavy_multiplication_x:       |  `/etc/pretix`                                                               | Path of the pretix config |
```
Comment (SlothOfAnarchy) [inline README.md]: typo `dor` to `dir`, maybe also in other places

File: README.md [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+| `pretix_dir`                     | :heavy_multiplication_x:       |  `/var/pretix`                                                               | Path where alle files will be stored |
+| `pretix_db_name`                     | :heavy_multiplication_x:       |  `pretix`                                                                 | Name of the Pretix DB (Postgres) |
+| `pretix_db_user`                     | :heavy_multiplication_x:       |  `pretix`                                                                 | Name of the Pretix DB User (Postgres) |
+| `pretix_redis_location`                     | :heavy_multiplication_x:       |  `127.0.0.1`                                                       | Host of the redis installation |
+
+
```
Comment (SlothOfAnarchy) [inline README.md]:
```suggestion
```

File: defaults/main.yml [hunk 9]
```diff
@@ -0,0 +1,25 @@
+---
+# Pretix
+pretix_user: pretix
+pretix_group: pretix
+pretix_currency: EUR
```
Comment (SlothOfAnarchy) [inline defaults/main.yml]: not described in README

File: handlers/main.yml [hunk 10]
```diff
@@ -0,0 +1,18 @@
+---
+
```
Comment (SlothOfAnarchy) [inline handlers/main.yml]:
```suggestion
```

File: handlers/main.yml [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+- name: Reload systemd daemon
+  systemd:
+    daemon_reload: yes
+  listen: "Restart pretix serivces"
```
Comment (SlothOfAnarchy) [inline handlers/main.yml]: FYI quoting is not necessary in this case

File: handlers/main.yml [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+
+
+- name: Restart pretix
+  service:
+    name: '{{ item }}'
+    state: restarted
+    enabled: yes
+  with_items:
```
Comment (SlothOfAnarchy) [inline handlers/main.yml]: `with_items` is deprecated, please use `loop` instead
```suggestion
  loop:
```

File: meta/main.yml [hunk 13]
```diff
@@ -0,0 +1,19 @@
+---
+galaxy_info:
+  author: Christophre Behrmann
```
Comment (SlothOfAnarchy) [inline meta/main.yml]: typo in your name

File: meta/main.yml [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
+  description: pretix instance
```
Comment (SlothOfAnarchy) [inline meta/main.yml]:
```suggestion
  description: Role to install the Pretix ticket sale system
```

File: tasks/main.yml [hunk 15]
```diff
@@ -0,0 +1,10 @@
+---
+- import_tasks: install.yml
+  tags: ['pretix-install']
```
Comment (SlothOfAnarchy) [inline tasks/main.yml]:
```suggestion
  tags: pretix-install
```
or
```suggestion
  tags:
    - pretix-install
```

same thing with the other tags

File: tasks/install.yml [hunk 16]
```diff
@@ -0,0 +1,87 @@
...
+  group:
+    name: '{{ pretix_group }}'
+    state: present
+
+- name: Create user
+  user:
+    name: '{{ pretix_user }}'
+    group: '{{ pretix_group }}'
+    state: present
+
```
Comment (SlothOfAnarchy) [inline tasks/install.yml]:
```suggestion
```

File: tasks/install.yml [hunk 17]
```diff
@@ -0,0 +1,87 @@
...
+    path: '{{ pretix_config_dir }}'
+    state: directory
+
+- name: Prepare directories
+  file:
+    path: '{{ item }}'
+    state: directory
+    owner: '{{ pretix_user }}'
+    group: '{{ pretix_group }}'
+#    mode: '0600'
```
Comment (SlothOfAnarchy) [inline tasks/install.yml]: why commented out? Perhaps `0640` or `0660` would be better

File: tasks/install.yml [hunk 18]
```diff
@@ -0,0 +1,87 @@
...
+  with_items:
```
Comment (SlothOfAnarchy) [inline tasks/install.yml]:
```suggestion
  loop:
```

File: tasks/install.yml [hunk 19]
```diff
@@ -0,0 +1,87 @@
...
+    - '{{ pretix_config_dir }}'
+    - '{{ pretix_dir }}/data/media'
+
+- name:  Create config file
+  template:
+    src: pretix.cfg.j2
+    dest: '{{ pretix_config_dir }}/pretix.cfg'
+    owner: '{{ pretix_user }}'
+    group: '{{ pretix_group }}'
+    mode: '0600'
```
Comment (SlothOfAnarchy) [inline tasks/install.yml]:
```suggestion
    mode: 0600
```

File: tasks/install.yml [hunk 20]
```diff
@@ -0,0 +1,87 @@
...
+
+- name: Install virtualenv via pip
+  pip:
+    name: virtualenv
+    executable: pip3
+
```
Comment (SlothOfAnarchy) [inline tasks/install.yml]:
```suggestion
```

File: tasks/install.yml [hunk 21]
```diff
@@ -0,0 +1,87 @@
...
+  become: true
+  become_user: '{{ pretix_user }}'
+  pip:
+    virtualenv: '{{ pretix_dir }}/venv'
+    name:
+      - setuptools
+      - wheel
+      - pretix[postgres]
+      - gunicorn
+
```
Comment (SlothOfAnarchy) [inline tasks/install.yml]:
```suggestion
```

File: tasks/install.yml [hunk 22]
```diff
@@ -0,0 +1,87 @@
...
+
+- name: Instll specific urllib
```
Comment (SlothOfAnarchy) [inline tasks/install.yml]:
```suggestion
- name: Install specific version of urllib
```

File: tasks/install.yml [hunk 23]
```diff
@@ -0,0 +1,87 @@
...
+    executable: pip3
+
+
+- name: Install python dependencies
+  become: true
+  become_user: '{{ pretix_user }}'
+  pip:
+    virtualenv: '{{ pretix_dir }}/venv'
+    name:
+      - setuptools
```
Comment (SlothOfAnarchy) [inline tasks/install.yml]: Are these packages not available via apt?

File: tasks/install.yml [hunk 24]
```diff
@@ -0,0 +1,87 @@
...
+
+- name: Instll specific urllib
+  become: true
+  become_user: '{{ pretix_user }}'
+  pip:
+    virtualenv: '{{ pretix_dir }}/venv'
+    name:
+      - urllib3==1.24
+
+
```
Comment (SlothOfAnarchy) [inline tasks/install.yml]: Remove trailing newlines

File: tasks/install.yml [hunk 25]
```diff
@@ -0,0 +1,87 @@
...
+    state: present
+
+- name: Create user
+  user:
+    name: '{{ pretix_user }}'
+    group: '{{ pretix_group }}'
+    state: present
+
+
+- name: Ensure working dir exists
```
Comment (SlothOfAnarchy) [inline tasks/install.yml]: Duplicate task to the next one

File: tasks/database.yml [hunk 26]
```diff
@@ -0,0 +1,15 @@
+---
+- name: Ensure database is created
+  postgresql_db:
+    name: "{{ pretix_db_name }}"
+
+- name: Ensure user has access to the database
+  postgresql_user:
```
Comment (SlothOfAnarchy) [inline tasks/database.yml]: Does this also create the user?

File: tasks/database.yml [hunk 27]
```diff
@@ -0,0 +1,15 @@
...
+    db: "{{ pretix_db_name }}"
+    name: "{{ pretix_db_user }}"
+    priv: ALL
```
Comment (SlothOfAnarchy) [inline tasks/database.yml]: Will these privileges apply only for this database or for all databases?

File: tasks/database.yml [hunk 28]
```diff
@@ -0,0 +1,15 @@
...
+
+- name: Ensure user does not have unnecessary privileges
+  postgresql_user:
+    name: "{{ pretix_db_user }}"
+    role_attr_flags: NOSUPERUSER,NOCREATEDB
```
Comment (SlothOfAnarchy) [inline tasks/database.yml]: Is it not possible to set in the task above?

File: templates/pretix.cfg.j2 [hunk 29]
```diff
@@ -0,0 +1,29 @@
+[pretix]
```
Comment (SlothOfAnarchy) [inline templates/pretix.cfg.j2]: Please insert a comment with the variable `ansible_managed`

File: templates/pretix.cfg.j2 [hunk 30]
```diff
@@ -0,0 +1,29 @@
...
+instance_name={{ pretix_instance_name }}
+url={{ pretix_instance_url }}
+currency={{ pretix_currency }}
+datadir={{ pretix_dir }}/data
+
+[database]
+backend=postgresql
+name={{ pretix_db_name }}
+user={{ pretix_user }}
+password=
```
Comment (SlothOfAnarchy) [inline templates/pretix.cfg.j2]: What about these keys? Shouldn't they have a value or be omitted when empty?

File: templates/pretix.cfg.j2 [hunk 31]
```diff
@@ -0,0 +1,29 @@
...
+host=
+
+[mail]
+from={{ pretix_mail_address }}
+host={{ pretix_mail_host }}
+user={{ pretix_mail_username }}
+password={{ pretix_mail_password }}
+port={{ pretix_mail_port }}
+tls={{ pretix_mail_tls }}
+
```
Comment (SlothOfAnarchy) [inline templates/pretix.cfg.j2]:
```suggestion
```

File: templates/pretix.cfg.j2 (hunk 29 above)
Comment (SlothOfAnarchy) [inline templates/pretix.cfg.j2]: This looks like a ini file. Maybe it would be possible to use the Ansible ini module. I don't know whether there are any advantages to it though.

File: templates/pretix_worker.service.j2 [hunk 32]
```diff
@@ -0,0 +1,15 @@
...
+User={{ pretix_user }}
+Group={{ pretix_group }}
+Environment="VIRTUAL_ENV={{ pretix_dir }}/venv"
+Environment="PATH={{ pretix_dir }}/venv/bin:/usr/local/bin:/usr/bin:/bin"
+ExecStart={{ pretix_dir }}/venv/bin/celery -A pretix.celery_app worker -l info
+WorkingDirectory={{ pretix_dir }}
+Restart=on-failure
+
+[Install]
+WantedBy=multi-user.target
```
Comment (SlothOfAnarchy) [inline templates/pretix_worker.service.j2]: Add missing linebreak

File: templates/pretix_worker.service.j2 [hunk 33]
```diff
@@ -0,0 +1,15 @@
+[Unit]
+Description=pretix background worker
```
Comment (SlothOfAnarchy) [inline templates/pretix_worker.service.j2]:
```suggestion
Description=Pretix background worker
```

File: templates/pretix_web.service.j2 [hunk 34]
```diff
@@ -0,0 +1,18 @@
...
+Environment="PATH=/venv/bin:/usr/local/bin:/usr/bin:/bin"
+ExecStart={{ pretix_dir }}/venv/bin/gunicorn pretix.wsgi \
+                      --name pretix --workers 5 \
+                      --max-requests 1200  --max-requests-jitter 50 \
+                      --log-level=info --bind=127.0.0.1:8345
+WorkingDirectory={{ pretix_dir }}
+Restart=on-failure
+
+[Install]
+WantedBy=multi-user.target
```
Comment (SlothOfAnarchy) [inline templates/pretix_web.service.j2]: Add missing linebreak

File: templates/pretix_web.service.j2 [hunk 35]
```diff
@@ -0,0 +1,18 @@
+[Unit]
+Description=pretix web service
+After=network.target
+
+[Service]
+User=pretix
```
Comment (SlothOfAnarchy) [inline templates/pretix_web.service.j2]: Please use the variables for `user` and `group`

File: templates/pretix_web.service.j2 [hunk 36]
```diff
@@ -0,0 +1,18 @@
+[Unit]
+Description=pretix web service
```
Comment (SlothOfAnarchy) [inline templates/pretix_web.service.j2]:
```suggestion
Description=Pretix web service
```

File: templates/pretix_web.service.j2 [hunk 37]
```diff
@@ -0,0 +1,18 @@
+[Unit]
```
Comment (SlothOfAnarchy) [inline templates/pretix_web.service.j2]: Please add an `ansible_managed` comment

File: templates/pretix_worker.service.j2 [hunk 38]
```diff
@@ -0,0 +1,15 @@
+[Unit]
```
Comment (SlothOfAnarchy) [inline templates/pretix_worker.service.j2]: Please add an `ansible_managed` comment

File: templates/pretix_web.service.j2 [hunk 39]
```diff
@@ -0,0 +1,18 @@
...
+After=network.target
+
+[Service]
+User=pretix
+Group=pretix
+Environment="VIRTUAL_ENV={{ pretix_dir }}/venv"
+Environment="PATH=/venv/bin:/usr/local/bin:/usr/bin:/bin"
+ExecStart={{ pretix_dir }}/venv/bin/gunicorn pretix.wsgi \
+                      --name pretix --workers 5 \
+                      --max-requests 1200  --max-requests-jitter 50 \
```
Comment (SlothOfAnarchy) [inline templates/pretix_web.service.j2]: It might be useful to make these values configurable

**Comment by Scriptkiddi** [comment]
@SlothOfAnarchy @astrocbxy is everything fixed?

**Comment by astrocbxy** [comment]
Sollte jetzt

File: README.md (hunk 2 above)
Comment (SlothOfAnarchy) [reply README.md]: .

File: defaults/main.yml [hunk 40]
```diff
@@ -0,0 +1,29 @@
...
+
+
+# Mail
+pretix_mail_host: 127.0.0.1
+pretix_mail_port: 587
+pretix_mail_tls: True
+
+# System service gunicorn
+pretix_gunicorn_workers: 5
+pretix_gunicorn_maxrequest: 1200
```
Comment (SlothOfAnarchy) [inline defaults/main.yml]: Missing newline

**Comment by SlothOfAnarchy** [comment]
@astrocbxy @Scriptkiddi ne aus dem Review fehlt noch einiges: https://github.com/stuvusIT/pretix/pull/1#pullrequestreview-230284141

**Comment by Scriptkiddi** [comment]
@astrocbxy wie siehts aus?

File: tasks/install.yml (hunk 17 above)
Comment (astrocbxy) [reply tasks/install.yml]: artefact of copying file creation task

File: tasks/install.yml (hunk 23 above)
Comment (astrocbxy) [reply tasks/install.yml]: they are probably, but we want to install them in our virtualenv not systemwide

File: tasks/database.yml (hunk 26 above)
Comment (astrocbxy) [reply tasks/database.yml]: yes

File: tasks/database.yml (hunk 27 above)
Comment (astrocbxy) [reply tasks/database.yml]: i think for this db

File: tasks/database.yml (hunk 28 above)
Comment (astrocbxy) [reply tasks/database.yml]: idk, this way it affects all db, in the taks above just the db "{{ pretix_db_name }}"

File: templates/pretix.cfg.j2 (hunk 30 above)
Comment (astrocbxy) [reply templates/pretix.cfg.j2]: https://docs.pretix.eu/en/latest/admin/installation/manual_smallscale.html
states, with a postgresql database passowd can be left blank if systemuser & db user have the same name

File: templates/pretix_worker.service.j2 (hunk 32 above)
Comment (astrocbxy) [reply templates/pretix_worker.service.j2]: can you explain that

File: templates/pretix.cfg.j2 (hunk 29 above)
Comment (astrocbxy) [reply templates/pretix.cfg.j2]: i don't think so

File: tasks/install.yml (hunk 23 above)
Comment (SlothOfAnarchy) [reply tasks/install.yml]: Why not system wide? Then they're maintained by the normal `apt` upgrade process

Comment (astrocbxy) [reply tasks/install.yml]: encapsulation in case of multiple installations of different softwares or different versions

File: README.md [hunk 41]
```diff
@@ -0,0 +1,109 @@
...
+    pretix_mail_name: stuvus - Tickets
+    pretix_mail_address: tickets@stuvus.uni-stuttgart.de
+    pretix_mail_username: ttickets
+    pretix_mail_password:  dasjkdajskdjaskjd
+    pretix_mail_host: mail01.faveve.uni-stuttgart.de
+
+    # Postgres
+    postgres_initdb: /usr/lib/postgresql/9.6/bin/initdb
+
```
Comment (SlothOfAnarchy) [inline README.md]:
```suggestion
```

File: README.md [hunk 42]
```diff
@@ -0,0 +1,109 @@
...
+
```
Comment (SlothOfAnarchy) [inline README.md]:
```suggestion
```

File: README.md [hunk 43]
```diff
@@ -0,0 +1,109 @@
...
+    # Nginx
+    served_domains:
+      - domains:
+          - tickets
+        privkey_path: /etc/ssl/privkey.pem
+        fullchain_path: /etc/ssl/fullchain.pem
+        default_server: true
+        crypto: true
+        https: false
+
```
Comment (SlothOfAnarchy) [inline README.md]:
```suggestion
```

File: README.md [hunk 44]
```diff
@@ -0,0 +1,109 @@
...
+          content: |
+            deny all;
+            return 404;
+        - condition: /static/ 
+          content: |
+            alias /var/pretix/venv/lib/python3.5/site-packages/pretix/static.dist/;
+            access_log off;
+            expires 365d;
+            add_header Cache-Control "public";
+
```
Comment (SlothOfAnarchy) [inline README.md]:
```suggestion
```

File: README.md [hunk 45]
```diff
@@ -0,0 +1,109 @@
...
+
```
Comment (SlothOfAnarchy) [inline README.md]:
```suggestion
```

File: handlers/main.yml [hunk 46]
```diff
@@ -0,0 +1,17 @@
+---
+- name: Reload systemd daemon
+  systemd:
+    daemon_reload: yes
+  listen: "Restart pretix serivces"
+
```
Comment (SlothOfAnarchy) [inline handlers/main.yml]:
```suggestion
```

File: handlers/main.yml [hunk 47]
```diff
@@ -0,0 +1,17 @@
...
+- name: Restart pretix
+  service:
+    name: '{{ item }}'
+    state: restarted
+    enabled: yes
+  loop:
+    - pretix_web
+    - pretix_worker
+  listen: "Restart pretix serivces"
+
```
Comment (SlothOfAnarchy) [inline handlers/main.yml]:
```suggestion
```

File: tasks/main.yml [hunk 48]
```diff
@@ -0,0 +1,10 @@
+---
+- import_tasks: install.yml
+  tags: pretix-install
+
+- import_tasks: database.yml
+  tags: pretix-database
+
+- import_tasks: configure.yml
+  tags: pretix-configure
+
```
Comment (SlothOfAnarchy) [inline tasks/main.yml]:
```suggestion
```

File: templates/pretix.cfg.j2 [hunk 49]
```diff
@@ -0,0 +1,29 @@
+# Ansible managed
```
Comment (SlothOfAnarchy) [inline templates/pretix.cfg.j2]:
```suggestion
# {{ ansible_managed }}
```

File: templates/pretix_web.service.j2 [hunk 50]
```diff
@@ -0,0 +1,19 @@
+#ansible_managed
```
Comment (SlothOfAnarchy) [inline templates/pretix_web.service.j2]:
```suggestion
# {{ ansible_managed }}
```

File: templates/pretix_worker.service.j2 [hunk 51]
```diff
@@ -0,0 +1,16 @@
+#ansible_managed
```
Comment (SlothOfAnarchy) [inline templates/pretix_worker.service.j2]:
```suggestion
# {{ ansible_managed }}
```

File: templates/pretix_worker.service.j2 (hunk 32 above)
Comment (SlothOfAnarchy) [reply templates/pretix_worker.service.j2]: You need to add a newline at the end of the file for the error to disappear.

File: tasks/database.yml [hunk 52]
```diff
@@ -0,0 +1,15 @@
+---
+- name: Ensure database is created
+  postgresql_db:
+    name: "{{ pretix_db_name }}"
+
+- name: Create datavbase user and  ensure the user access to the database
```
Comment (SlothOfAnarchy) [inline tasks/database.yml]:
```suggestion
- name: Create database user and ensure the user access to the database
```

File: tasks/install.yml [hunk 53]
```diff
@@ -0,0 +1,74 @@
...
+  become_user: '{{ pretix_user }}'
+  pip:
+    virtualenv: '{{ pretix_dir }}/venv'
+    name:
+      - setuptools
+      - wheel
+      - pretix[postgres]
+      - gunicorn
+
+- name: Instll specific version of urllib
```
Comment (SlothOfAnarchy) [inline tasks/install.yml]:
```suggestion
- name: Install specific version of urllib
```

File: README.md [hunk 54]
```diff
@@ -0,0 +1,104 @@
...
+## Example Playbook
+
+```yml
+- hosts: all
+  become: true
+  vars:
+    pretix_instance_name: Stuvus Tickettest
+    pretix_instance_url: tickets.int.stuvus.uni-stuttgart.de
+
```
Comment (SlothOfAnarchy) [inline README.md]:
```suggestion
```

File: README.md [hunk 55]
```diff
@@ -0,0 +1,104 @@
...
+            expires 365d;
+            add_header Cache-Control "public";
+
+  roles:
+    - redis
+    - postgres
+    - nginx
+    - pretix
+```
+
```
Comment (SlothOfAnarchy) [inline README.md]:
```suggestion


```

File: defaults/main.yml [hunk 56]
```diff
@@ -0,0 +1,29 @@
...
+
+# Redis
+pretix_redis_location: 127.0.0.1
+
+
+# Mail
+pretix_mail_host: 127.0.0.1
+pretix_mail_port: 587
+pretix_mail_tls: True
+
```
Comment (SlothOfAnarchy) [inline defaults/main.yml]:
```suggestion


```
